src/neo/SmartContract/Native/StdLib.cs

```diff
@@ -135,9 +137,48 @@ public static string Base58Encode(byte[] data)
         /// <param name="s">The base58 <see cref="string"/>.</param>
         /// <returns>The decoded byte array.</returns>
         [ContractMethod(CpuFee = 1 << 12)]
-        public static byte[] Base58Decode(string s)
+        public static byte[] Base58Decode([MaxLength(MaxInputLength)] string s)
         {
             return Base58.Decode(s);
         }
+
+        [ContractMethod(CpuFee = 1 << 12)]
+        private static int MemoryCompare([MaxLength(MaxInputLength)] byte[] str1, [MaxLength(MaxInputLength)] byte[] str2)
+        {
+            return Math.Sign(str1.AsSpan().SequenceCompareTo(str2));
+        }
+
+        [ContractMethod(CpuFee = 1 << 12)]
+        private static int MemorySearch(byte[] mem, byte[] value)
+        {
+            return MemorySearch(mem, value, 0, false);
+        }
+
+        [ContractMethod(CpuFee = 1 << 12)]
+        private static int MemorySearch(byte[] mem, byte[] value, int start)
+        {
+            return MemorySearch(mem, value, start, false);
+        }
+
+        [ContractMethod(CpuFee = 1 << 12)]
+        private static int MemorySearch([MaxLength(MaxInputLength)] byte[] mem, byte[] value, int start, bool backward)
+        {
+            if (backward)
+            {
+                return mem.AsSpan(0, start).LastIndexOf(value);
+            }
+            else
+            {
+                int index = mem.AsSpan(start).IndexOf(value);
+                if (index < 0) return -1;
+                return index + start;
+            }
+        }
+
+        [ContractMethod(CpuFee = 1 << 12)]
+        private static string[] StringSplit([MaxLength(MaxInputLength)] string str, string separator)
```

Comment:
Good string operations. This class was a good move and, in summary, the migration to Native operations.